### PR TITLE
Minor Changes and Adding Multiple Courses to Contacts

### PR DESCRIPTION
**Pull request by jahnaviraj**

### CODE REVIEW

File: desk/src/pages/desk/Desk.vue [hunk 1]
```diff
@@ -562,6 +615,16 @@ export default {
 				},
 			}
 		},
+		assignContactCourse() {
+			return {
+				method: 'frappedesk.api.ticket.assign_contact_course',
+				onSuccess: async (contact) => {},
+				onError: (error) => {
+					console.log(error)
```
Comment (ashishnpbs) [inline desk/src/pages/desk/Desk.vue:623]: instead of console.log shall we have frappe.log_error?

File: desk/src/pages/desk/Desk.vue [hunk 2]
```diff
@@ -571,6 +634,15 @@ export default {
 				}
 			}
 		},
+		deleteContactCourse() {
+			return {
+				method: 'frappedesk.api.ticket.delete_contact_course',
+				onSuccess: async (contact) => {},
+				onError: (error) => {
+					console.log(error)
```
Comment (ashishnpbs) [inline desk/src/pages/desk/Desk.vue:642]: same here

File: desk/src/pages/desk/Desk.vue [hunk 3]
```diff
@@ -627,6 +699,18 @@ export default {
...
 		},
+		createContactCourse() {
+			return {
+				method: 'frappedesk.api.ticket.check_and_create_contact_course',
+				onSuccess: () => {
+					this.$resources.courses.fetch()
+				},
+				onError: (error) => {
+					console.log(error)
+					// TODO:
```
Comment (ashishnpbs) [inline desk/src/pages/desk/Desk.vue:710]: same here

File: frappedesk/frappedesk/doctype/contact_course/contact_course.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
+{
```
Comment (ashishnpbs) [inline frappedesk/frappedesk/doctype/contact_course/contact_course.json:1]: permissions?

File: desk/src/components/desk/ticket/InfoPanel.vue [hunk 5]
```diff
@@ -33,7 +33,66 @@
...
+									@click="() => {
+										this.openCreateNewContactCourseDialog = true
+									}"
+								>
+									Create new
+								</div>
+							</template>
+						</Autocomplete>
+						</div> 
+						<div v-if="ticket.contact.department" class="flex flex-row items-center space-x-[12px]">
```
Comment (ashishnpbs) [inline desk/src/components/desk/ticket/InfoPanel.vue:95]: department ? is it for course or actual department

File: desk/src/pages/desk/Desk.vue (hunk 1 above)
Comment (jahnaviraj) [reply desk/src/pages/desk/Desk.vue:623]: I dont think that is accessibile in vue js

File: frappedesk/frappedesk/doctype/contact_course/contact_course.json (hunk 4 above)
Comment (jahnaviraj) [reply frappedesk/frappedesk/doctype/contact_course/contact_course.json:1]: It is child table, does not have permissions. Permissions for this is same as for its parent.

File: desk/src/components/desk/ticket/InfoPanel.vue (hunk 5 above)
Comment (jahnaviraj) [reply desk/src/components/desk/ticket/InfoPanel.vue:95]: Removed, added new component for displaying courses

**Pull request merged**